mojo-executor-maven-plugin/src/main/java/org/twdata/maven/mojoexecutor/plugin/MojoExecutorMojo.java

```diff
@@ -83,13 +89,47 @@
      */
     private BuildPluginManager pluginManager;
 
-    public void execute() throws MojoExecutionException {
-        if ( mavenProject.getArtifactId().equals( "mojo-executor-test-project-null-maven-project" ))
+    public void execute() throws MojoExecutionException
+    {
+        integrationTestSetup();
+
+        getLog().info( "Executing with maven project " + mavenProject + " for session " + mavenSession );
+        executeMojo( plugin, goal, toXpp3Dom( configuration ),
+                     executionEnvironment( mavenProject, mavenSession, pluginManager ) );
+    }
+
+    /**
+     * Used purely to customise the setups for the integration tests.
+     *
+     * @throws MojoExecutionException
+     */
+    private void integrationTestSetup() throws MojoExecutionException
+    {
+        // Test specific conditionals...
```

Comment:
I see what you're saying ; however essentially the relevant integration tests need custom plugin to verify the changes work as expected. I'll have a think to see if there could be a better layout to apply.